homeworks/vlad.makedon_Vladmaked/homework_6/horses-race.js

```diff
@@ -0,0 +1,78 @@
+const intervalIds = [];
+
+const horseNames = ['Pecos', 'Colt', 'Horseshoe', 'Barley', 'Bronco', 'Blazer', 'Wilbur', 'Buck',
+    'Franklin', 'Kentucky', 'Trigger', 'Blacksmith', 'Buckeye', 'Bolt', 'Washington', 'Connecticut',
+    'Ironclad', 'King', 'Jockey', 'Clydesdale', 'Wrangler', 'Jupiter', 'Tennessee', 'Texas', 'Diesel',
+    'Tank', 'Colorado', 'Ridgeline', 'Quarterback', 'Shoelace', 'Colonel', 'Lincoln', 'Steeltoe', 'Officer',
+    'Rushmore', 'Ellie', 'Peanut', 'River', 'Marmalade', 'Nutmeg', 'Piper', 'Gypsy', 'Butterscotch', 'Clover',
+    'Sadie', 'Sassy', 'Bailey', 'Skydancer', 'Daisy', 'Strawberry', 'Kennedy', 'Morgan', 'Nightmare', 'Magnolia',
+    'Mustang Sally', 'Hollyhock', 'Aster', 'Molly', 'Willow', 'Meadow', 'Ladybird', 'Carolina', 'Wisteria', 'Cherish',
+    'Rosette', 'Midsummer', 'Amethyst', 'Eleanor', 'Juliet', 'Jasmine', 'Velvet', 'Big Mama', 'Temperance'];
+const horseBreeds = ['Thoroughbred', 'American Quarter Horse', 'Arabian Horse', 'Paint Horse', 'Appaloosa'];
+
+class Horse {
+    constructor(name, breed) {
+        this.name = name;
+        this.breed = breed;
+    }
+}
+
+class Racer extends Horse {
+    constructor(name, breed) {
+        super(name, breed);
+        this.distance = 0;
+        this.setSpeed();
+    }
+
+    run() {
+        intervalIds.push(setInterval(() => {
```

Comment:
pushing to array in global scope is not welcomed unless it's really needed, it this case intervals are related to `Racer` and `Race` classes, so you could store this info in class properties